pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -18,96 +18,92 @@ type ServiceBindingRequestSpec struct {
 	// +optional
 	EnvVarPrefix string `json:"envVarPrefix,omitempty"`
 
-	// Custom env variables
+	// DataMapping is used to map CR/CSV/CRD attributes to Custom env variables
 	// +optional
-	CustomEnvVar []corev1.EnvVar `json:"customEnvVar,omitempty"`
-
-	// BackingServiceSelector is used to identify the backing service operator.
-	// Deprecation Notice:
-	// In the upcoming release, this field would be depcreated. It would be mandatory
-	// to set "backingServiceSelectors".
-	// +optional
-	BackingServiceSelector *BackingServiceSelector `json:"backingServiceSelector,omitempty"`
+	DataMapping []corev1.EnvVar `json:"dataMapping,omitempty"`
 
 	// BackingServiceSelectors is used to identify multiple backing services.
-	// This would be made a required field after 'BackingServiceSelector'
+	// This would be made a required field after 'services'
 	// is removed.
-	// +optional
-	BackingServiceSelectors *[]BackingServiceSelector `json:"backingServiceSelectors,omitempty"`
+	Services []Service `json:"services"`
```

Comment:
`Service` highlights just the backing service when in fact, both `Service` and `Application` are an integral part of `Service Binding Operator`.

There cannot be a "binding" without atleast one of each - `Service` and `Application`.

Are we saying that `Service` is THE MOST IMPORTANT component of `Service Binding Operator`?

Reply:
We could do a generic binding without specifying `Application` ie, only specify `Services`. However, if we don't provide `Services`, we wouldn't have anything to add to the binding secret.